compat/getgrent_r.c

```diff
@@ -37,6 +37,16 @@
 #include <errno.h>
 #include <string.h>
 
+#if defined(BSD)
+#include <sys/param.h>
+#else
+// This branch is necessarily Linux and not GNU because of the checks
+// defined above that guard the rest of the compat layer.  On Linux we
+// don't pull in param.h as it is very obsolete.
+#include <stdint.h>
+#define ALIGNBYTES _Alignof(max_align_t)
```

Comment:
This type requires C11 which isn't a real problem but the symbols might not be visible depending on the system.  What do you think about removing the ifdefs and param.h include altogether and doing this:
```
#define ALIGNBYTES (sizeof(uintptr_t) - 1)
```

Reply:
I had considered the C11 issue when drafting this patch but thought it shouldn't be an issue on FreeBSD or most modern *nix's.  I'm all for removing param.h dependencies, so if you are confident that changing from the _Alignof won't affect any currently working target I'd say go ahead and send the PR.